app/src/test/java/com/rkm/tasky/network/authentication/implementation/AuthenticationManagerImplTest.kt

```diff
@@ -1,70 +1,161 @@
+        val password = "password"
+        val fullName = "Bob Smith"
+        val result = authManager.registerUser(fullName, email, password)
+        assertTrue(result is Result.Success)
+    }
+
+    @Test
+    fun `registration failed`() = runTest(dispatcher) {
+        val email = "email@gmail.com"
+        val password = "password"
+        val fullName = "Bob Smith"
+        authRepository.setHasError(true)
+        val result = authManager.registerUser(fullName, email, password)
+        assertTrue(result is Result.Error)
+    }
+
+    @Test
+    fun `check authentication success`() = runTest(dispatcher) {
+        val result = authManager.checkAuthentication()
+        assertTrue(result is Result.Success)
+    }
+
+    @Test
+    fun `check authentication failed`() = runTest(dispatcher) {
+        authRepository.setHasError(true)
+        val result = authManager.checkAuthentication()
+        assertTrue(result is Result.Error)
+    }
+
+    @Test
+    fun `get new access token session empty`() = runTest {
+        val result = authManager.getNewAccessToken()
+        assertTrue(result is Result.Error)
+        assertEquals((result as Result.Error).error, NetworkError.APIError.UNKNOWN)
+    }
+
+    @Test
+    fun `get new access token session success`() = runTest {
+        val email = "email@gmail.com"
+        val password = "password"
+        val authInfo = loginUserResponseToPojo().asLoginDTO().asAuthInfo()
+        val tokenResponse = getNewAccessTokenResponseToPojo().asAccessTokenDTO()
+        val updatedAuthInfo = authInfo.copy(
+            accessToken = tokenResponse.accessToken,
+            accessTokenExpirationTimestamp = tokenResponse.expirationTimestamp
+        )
+        val result = authManager.logIn(email, password)
+        assertTrue(result is Result.Success)
+        assertTrue(sessionStorage.getSession() != null)
+        val result2 = authManager.getNewAccessToken()
+        assertTrue(result2 is Result.Success)
+        assertEquals(sessionStorage.getSession(), updatedAuthInfo)
+    }
+
+    @Test
+    fun `get new access token session failed`() = runTest {
+        val email = "email@gmail.com"
+        val password = "password"
+        val authInfo = loginUserResponseToPojo().asLoginDTO().asAuthInfo()
+        val tokenResponse = getNewAccessTokenResponseToPojo().asAccessTokenDTO()
+        val updatedAuthInfo = authInfo.copy(
+            accessToken = tokenResponse.accessToken,
+            accessTokenExpirationTimestamp = tokenResponse.expirationTimestamp
+        )
+        val result = authManager.logIn(email, password)
+        assertTrue(result is Result.Success)
+        assertTrue(sessionStorage.getSession() != null)
+        authRepository.setHasError(true)
+        val result2 = authManager.getNewAccessToken()
+        assertTrue(result2 is Result.Error)
+        assertNotEquals(sessionStorage.getSession(), updatedAuthInfo)
+        assertEquals(sessionStorage.getSession(), authInfo)
```

Comment:
I'd highly recommend to only test one thing per test. If a test breaks you want to know what broke. If this one breaks, you have no idea if it's the `logIn()` or `getNewAccessToken()` function and have to further investigate (while not getting the checks for the rest that may be successful)